src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -265,12 +260,13 @@ void ReadFromMergeTree::AnalysisResult::checkLimits(const Settings & settings, c
 ReadFromMergeTree::ReadFromMergeTree(
     MergeTreeData::DataPartsVector parts_,
     std::vector<AlterConversionsPtr> alter_conversions_,
+    const Names & column_names_,
```

Comment:
Why do we need column_names_ here now?

Reply:
`SourceStepWithFilter` now becomes a common base plan step for all plan nodes which needs key analysis and prewhere. It needs to store queried source columns for `MergeTreeWhereOptimizer`. In old interpreter, this info is grabbed from TreeRewriterResult, but we don't have that in our modern optimizer.